src/json-helper.c

```diff
@@ -27,6 +27,51 @@
 #include "json-helper.h"
 #include <stddef.h>
 
+
+/**
+ * @brief Get the first JsonNode element matching path in json_node.
+ *
+ * @param[in]  json_node JsonNode to query
+ * @param[in]  path      Query path
+ * @param[out] error     Error
+ * @return JsonNode*, matching JsonNode element (must be freed), NULL on error
+ */
+static JsonNode* json_get_first_matching_element(JsonNode *json_node,
+                                                 const gchar *path,
+                                                 GError **error)
+{
+        g_autoptr(JsonNode) match = NULL, node = NULL;
+        JsonArray *arr = NULL;
+
+        g_return_val_if_fail(error == NULL || *error == NULL, NULL);
```

Comment:
please do function argument checks in the order they appear in the function signature

Reply:
Done.